src/codegen.md

```diff
@@ -0,0 +1,64 @@
+# Code generation
+
+<!-- TODO: target_feature -->
+
+## Optimization hints
+
+The `cold` and `inline` [attributes] give suggestions to the compiler to
```

Comment:
Absolutely should be in second person. I mostly tried to avoid editing the existing documentation, since I wanted to focus on just reorganization and adding the built-in syntax.  I also agree with the compiler/interpreter neutrality.